homeassistant/components/beewi_smartclim/sensor.py

```diff
@@ -0,0 +1,187 @@
+"""Platform for beewi_smartclim integration."""
+import logging
+
+from datetime import datetime, timedelta
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_MONITORED_CONDITIONS,
+    CONF_MAC,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_BATTERY,
+)
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+# Default values
+DEFAULT_NAME = "BeeWi SmartClim"
+
+# Sensor config
+SENSOR_TYPES = {
+    "temperature": [DEVICE_CLASS_TEMPERATURE, "Temperature", TEMP_CELSIUS],
+    "humidity": [DEVICE_CLASS_HUMIDITY, "Humidity", "%"],
+    "battery": [DEVICE_CLASS_BATTERY, "Battery", "%"],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_MAC): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+        ),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the beewi_smartclim platform."""
+    _LOGGER.info("BeeWi SmartClim has been loaded")
+
+    mac = config.get(CONF_MAC)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.